### PR TITLE
Leave chat

### PR DESCRIPTION
**Pull request by sonamdkindy**
functionality for leaving a chat
- added icon for leaving chat in header (to the right of add user icon)
- chats added to props along with removeUserFromChat action func
- alert confirmation upon clicking button/icon to leave chat (you can hit cancel to close the alert and view the chat screen once again - else OK confirms you want to leave chat so it will call the removeUserFromChat action func)
- added removeUserFromChat action func and delete user from chat action type
- updated reducer with delete user from chat action type, such that chats is updated (using .filter to return new chats array filtering out the chat with same id of the current chat)
- componentDidUpdate hook so that i can show a toast shown upon successful adding of a user to chat _and_ leaving chat (checking for the appropriate changes - former: addedUser; latter: chats)
- redirect to chats if leaving chat (and chat if adding user in previous pr)

let me know if i missed anything or there's something i can improve!

### CODE REVIEW

File: reducers/chatsReducer.js [hunk 1]
```diff
@@ -26,26 +26,31 @@ function chatsReducer(state = chatsInitialState, action) {
...
+            const cidIndex = state.chats.findIndex(
+                (chat) => chat.id === cid
             );
             return {
                 ...state,
                 addedUser: user,
                 chats: {
                     ...state.chats,
-                    [chats[cidIndex].users]: [
+                    [state.chats[cidIndex].users]: [
```
Comment (mnicole) [inline reducers/chatsReducer.js]: i think i missed this last time... it's updating the state to be like this:
Screen Shot 2019-05-19 at 5 53 53 PM

could do something like
```
const chatToUpdate = Object.assign({}, state.chats[cidIndex]);
const newUsers = { users: [...chatToUpdate.users, user] };
const newChat = {...chatToUpdate, ...newUsers};
return {
                ...state,
                addedUser: user,
                chats: {
                    ...state.chats,
                    [cidIndex]: newChat
                }
            };
```

File: screens/AddUserToChatScreen.js [hunk 2]
```diff
@@ -101,9 +101,15 @@ class AddUserToChatScreen extends React.Component {
...
     componentDidUpdate(prevProps) {
-        // if adding user successful, navigate back to chat
+        // if adding user successful, display toast and navigate back to chat
         if (prevProps.addedUser !== this.props.addedUser) {
-            this.props.navigation.navigate('Chat');
+            Toast.show({
+                text: `${this.props.addedUser.displayName} added to chat`,
+                buttonText: "Okay",
+                type: "success",
+                duration: 1000,
```
Comment (mnicole) [inline screens/AddUserToChatScreen.js]: this is just personal preference but i'd make this a higher number because it's hard to read in 1 second

**Comment by mnicole** [comment]
looks awesome! toasts look nice :)

File: reducers/chatsReducer.js (hunk 1 above)
Comment (sonamdkindy) [reply reducers/chatsReducer.js]: Yikes, I knew there was probably something wrong with what he added previously. I might've mucked it up further too, lmao. I'll push up these changes when I get back home!

File: screens/AddUserToChatScreen.js (hunk 2 above)
Comment (sonamdkindy) [reply screens/AddUserToChatScreen.js]: I'll prob make it 2 seconds 'cause when I tried out 3 it felt way too long. Lmk if you think differently!

File: reducers/chatsReducer.js [hunk 3]
```diff
@@ -14,38 +14,56 @@ function chatsReducer(state = chatsInitialState, action) {
...
-                    [chats[cidIndex].users]: [
-                        ...state.chats[cidIndex].users,
-                        user
-                    ]
-                }
+                chats: [
+                    ...state.chats.slice(0, cidIndex),
+                    updatedChat,
+                    ...state.chats.slice(cidIndex + 1)
+                ]
```
Comment (sonamdkindy) [inline reducers/chatsReducer.js]: yeah, so another thing i noticed is chats are not the right datatype :/ and that originated in [his previous pr](https://github.com/emote-chat/client/pull/10/files). updated so it's an array and i followed the [documentation](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns) on how to update the chat at the given index. i wanted to use map here with a ternary operator, but i wasn't sure what was general/best practice here from a react standpoint. lmk if you have any particular opinion either way.

File: reducers/chatsReducer.js [hunk 4]
```diff
@@ -14,38 +14,54 @@ function chatsReducer(state = chatsInitialState, action) {
...
-                chats: {
-                    ...state.chats,
-                    [chats[cidIndex].users]: [
-                        ...state.chats[cidIndex].users,
-                        user
-                    ]
-                }
+                chats: state.chats.map(
+                    (chat, i) => i === cidIndex ? updatedChat : chat
+                )
```
Comment (sonamdkindy) [inline reducers/chatsReducer.js]: heads up, i changed this so it's using .map instead of using slicing. it just looks like a lot cleaner imo.

**Pull request merged**